### PR TITLE
Added gcd implementation [C]

### PR DESCRIPTION
**Pull request by Monal5031**
Added gcd implementation [C]

### CODE REVIEW

File: euclidean_gcd/euclidean_gcd.c [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+		
+}
+
+int main() {
+        int first = 49;
+        int second = 7;
+        int answer = gcd_recursive(first, second);
+
+        printf("GCD of %d and %d is : %d\n",first,second,answer);
+}
```
Comment (singhpratyush) [inline euclidean_gcd/euclidean_gcd.c]: Check results for both implementations.

File: euclidean_gcd/euclidean_gcd.c [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+int gcd_iterative(int first, int second) {
+		while(second ! = 0) {			// Iterate till second becomes zero
+			int temp = second;			// temporary variable to hold value of second which is to be assigned to first later
+			second = first % second;
+			first = temp;
+		}
+
+		return first;			// When second becomes 0, first becomes gcd of both
+
```
Comment (singhpratyush) [inline euclidean_gcd/euclidean_gcd.c]: Remove extra blank lines.

File: euclidean_gcd/euclidean_gcd.c [hunk 3]
```diff
@@ -0,0 +1,36 @@
+#include<stdio.h>
+
+/*
+* first --> First number
+* second --> Second number
+*
+* There are two implementations: Recursive (gcd_recursive) and Non-Recursive (gcd_iterative)
+*
```
Comment (singhpratyush) [inline euclidean_gcd/euclidean_gcd.c]: Comment lines are too much.

File: euclidean_gcd/euclidean_gcd.c [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        return first;           // When second becomes 0, first becomes gcd of both
+}
+
+int main() {
+        int first = 49;
+        int second = 7;
+        int answer = gcd_recursive(first, second);
+
+        printf("GCD of %d and %d is : %d by recursive algo.\n",first,second,answer);
+        printf("GCD of %d and %d is : %d by Iterative algo.\n",first,second,answer);
```
Comment (singhpratyush) [inline euclidean_gcd/euclidean_gcd.c]: * How exactly will this calculate GCD using iterative algo without calling the corresponding method?
* Add a space after comma - `..first, second, ...`.

Comment (Monal5031) [inline euclidean_gcd/euclidean_gcd.c]: Oh Sorry, I forgot to call the function.

File: euclidean_gcd/euclidean_gcd.c [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+*/
+
+int gcd_recursive(int first, int second) {
+        if(second == 0)
+            return first;                //first becomes gcd if second becomes zero
+        else
+            return gcd_recursive(second, (first % second));
+}
+
+int gcd_iterative(int first, int second) {
```
Comment (singhpratyush) [inline euclidean_gcd/euclidean_gcd.c]: The filename and method name needs to be the same.

Comment (Monal5031) [inline euclidean_gcd/euclidean_gcd.c]: But I had to differentiate the recursive and iterative implementation thats why I named them gcd_recursive and gcd_iterative.
Shall I make it euclidean_gcd_recursive and euclidean_gcd_iterative?

Comment (singhpratyush) [inline euclidean_gcd/euclidean_gcd.c]: How about `euclidean_gcd` and `euclidean_gcd_recursive`?

Comment (Monal5031) [inline euclidean_gcd/euclidean_gcd.c]: Okay I'll make the changes.

File: README.md [hunk 6]
```diff
@@ -11,6 +11,7 @@ Community (college) maintained list of algorithm implementations and codes.
 | [Linear Search](https://en.wikipedia.org/wiki/Linear_search) | | | | [\[X\]](linear_search/linear_search.py) |
 | [Longest Common Subsequence](http://www.geeksforgeeks.org/dynamic-programming-set-4-longest-common-subsequence) | [\[X\]](longest_common_subsequence/longestCommonSubsequence.c) | | | |
 | [Quicksort](https://en.wikipedia.org/wiki/Quicksort) | [\[X\]](quicksort/quicksort.c) | | | |
+| [Euclidean GCD](https://en.wikipedia.org/wiki/Euclidean_algorithm) | [\[X\]](euclidean_gcd/euclidean_gcd.c) | | | |
```
Comment (singhpratyush) [inline README.md]: Keep them in ascending order.

File: euclidean_gcd/euclidean_gcd.c [hunk 7]
```diff
@@ -0,0 +1,32 @@
+#include<stdio.h>
+
+/*
+* first --> First number
+* second --> Second number
+* There are two implementations: Recursive (gcd_recursive) and Non-Recursive (gcd_iterative)
```
Comment (singhpratyush) [inline euclidean_gcd/euclidean_gcd.c]: Please correct this comment.

File: euclidean_gcd/euclidean_gcd.c [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+        return first;           // When second becomes 0, first becomes gcd of both
+}
+
+int main() {
+        int first = 49;
+        int second = 7;
+        int answer_recursive = euclidean_gcd_recursive(first, second);
+        int answer_iterative = euclidean_gcd(first, second);
+        printf("GCD of %d and %d is : %d by recursive algo.\n", first, second, answer_recursive);
+        printf("GCD of %d and %d is : %d by Iterative algo.\n", first, second, answer_iterative);
```
Comment (singhpratyush) [inline euclidean_gcd/euclidean_gcd.c]: `I` should be in small case.

**Comment by singhpratyush** [comment]
Thanks!

**Pull request merged**